table/pit-cs.go

```diff
@@ -8,56 +8,86 @@
 package table
 
 import (
-	"bytes"
-	"math/rand"
 	"time"
 
-	"github.com/cespare/xxhash"
-	"github.com/named-data/YaNFD/core"
 	"github.com/named-data/YaNFD/ndn"
 )
 
-// PitCs represents the PIT-CS tree for a thread.
-type PitCs struct {
-	root               *PitCsNode
-	ExpiringPitEntries chan *PitEntry
-
-	nPitEntries int // Number of PIT entries in tree
-	pitTokenMap map[uint32]*PitEntry
-
-	nCsEntries    int // Number of CS entries in tree
-	csReplacement CsReplacementPolicy
-	csMap         map[uint64]*CsEntry
-}
-
-// PitCsNode represents an entry in a PIT-CS tree.
-type PitCsNode struct {
-	component ndn.NameComponent
-	depth     int
-
-	parent   *PitCsNode
-	children []*PitCsNode
-
-	pitEntries []*PitEntry
-
-	csEntry *CsEntry
-}
-
-// PitEntry is an entry in a thread's PIT.
-type PitEntry struct {
-	node  *PitCsNode
-	pitCs *PitCs
-
-	Name           *ndn.Name
-	CanBePrefix    bool
-	MustBeFresh    bool
-	ForwardingHint *ndn.Name                // Interests must match in terms of Forwarding Hint to be aggregated in PIT.
-	InRecords      map[uint64]*PitInRecord  // Key is face ID
-	OutRecords     map[uint64]*PitOutRecord // Key is face ID
-	ExpirationTime time.Time
-	Satisfied      bool
-
-	Token uint32
+// PitCsTable dictates what functionality a Pit-Cs table should implement
+type PitCsTable interface {
+	InsertInterest(interest *ndn.Interest, hint *ndn.Name, inFace uint64) (PitEntry, bool)
```

Comment:
I read that the best practice in Go is to "accept interfaces, return structs." However, here the function declaration says that we're returning an interface (of type `PitEntry`). In actual implementations, a real struct that implements `PitEntry` would be returned, so is this type of function declaration acceptable? Or is there a better way to do it?